src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/PackageUpdateResource.cs

```diff
@@ -563,5 +607,37 @@ public SearchPathResult(string path, bool isFile)
                 _isFile = isFile;
             }
         }
+
+        private bool IsV2LocalRepository(string root)
+        {
+            if (!Directory.Exists(root) ||
+                Directory.GetFiles(root, "*.nupkg").Any())
+            {
+                // If the repository does not exist or if there are .nupkg in the path, this is a v2-style repository.
+                return true;
+            }
+ 
+            foreach (var idDirectory in Directory.GetDirectories(root))
+            {
+                if (Directory.GetFiles(idDirectory, "*.nupkg").Any() ||
+                    Directory.GetFiles(idDirectory, "*.nuspec").Any())
+                {
+                    // ~/Foo/Foo.1.0.0.nupkg (LocalPackageRepository with PackageSaveModes.Nupkg) or 
+                    // ~/Foo/Foo.1.0.0.nuspec (LocalPackageRepository with PackageSaveMode.Nuspec)
+                    return true;
+                }
+                var idDirectoryName = Path.GetFileName(idDirectory);
+                foreach (var versionDirectoryPath in Directory.GetDirectories(idDirectory))
+                {
+                    if (Directory.GetFiles(versionDirectoryPath, idDirectoryName + NuGetConstants.ManifestExtension).Any())
+                    {
+                        // If we have files in the format {packageId}/{version}/{packageId}.nuspec, assume it's an expanded package repository.
+                        return false;
```

Comment:
i ported from nuget2 strictly, for the sole backward compatibility reason, as i have no context